### PR TITLE
feat(hx-search-control): add event listener and component tests

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Test Case `<hx-search-control>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2062

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-search-control/index.js [hunk 1]
```diff
@@ -13,21 +14,20 @@ export class HXSearchControlElement extends HXFormControlElement {
         return 'hx-search-control';
     }
 
-    $onCreate () {
```
Comment (100stacks) [inline src/elements/hx-search-control/index.js:16]: Was there an error that caused this to be removed?

File: src/elements/hx-search-control/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            const component = /** @type {HXSearchControlElement} */ await fixture(template);
+            const shadow = component.shadowRoot;
+
+            expect(shadow).to.be.null;
+        });
+    });
+
+    describe('test for event listeners', () => {
+        it('should fire a input event', async () => {
+            const mockup = `<hx-search-control>
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            setTimeout(() => component.dispatchEvent(customEvent));
+            const evt = await oneEvent(component, 'input');
+
+            expect(evt).to.equal(customEvent);
+            expect(evt.detail).to.equal(detail);
+        });
+
+        it('should fire a click event', async () => {
+            const mockup = `<hx-search-control>
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.js [hunk 4]
```diff
@@ -65,6 +65,12 @@ export class HXSearchControlElement extends HXFormControlElement {
         this._btnReset.hidden = true;
     }
 
+    /** @private */
+    _onHxchange (evt) {
+        evt.stopPropagation();
+        this.$defaultAttribute(STATE.changed, '');
+    }
+
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.js (hunk 1 above)
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.js:16]: Yes James, it throws an error while running yarn test "Error: Uncaught TypeError: Cannot read property 'bind' of undefined"

File: src/elements/hx-search-control/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            const component = /** @type {HXSearchControlElement} */ await fixture(template);
+            const shadow = component.shadowRoot;
+
+            expect(shadow).to.be.null;
+        });
+    });
+
+    describe('test for event listeners', () => {
+        it('should fire a input event', async () => {
+            const component = /** @type { HXSearchControlElement } */ await fixture(mockup);
```
Comment (100stacks) [inline src/elements/hx-search-control/index.spec.js:61]: Please change the variable from `component` to `fragment` to help signify it's a more complex structure.

File: src/elements/hx-search-control/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+            setTimeout(() => component.dispatchEvent(customEvent));
+            const evt = await oneEvent(component, 'input');
+
+            expect(evt).to.equal(customEvent);
+            expect(evt.detail).to.equal(detail);
+        });
+
+        it('should fire a click event', async () => {
+            const component = /** @type { HXSearchControlElement } */ await fixture(mockup);
```
Comment (100stacks) [inline src/elements/hx-search-control/index.spec.js:73]: Please change the variable from `component` to `fragment` to help signify it's a more complex structure.

File: src/elements/hx-search-control/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+/**
+ * <hx-search-control> component tests
+ *
+ * @type HXSearchControlElement
+ *
+ */
+describe('<hx-search-control> component tests', () => {
+    const template = '<hx-search-control>';
+    const mockup = `<hx-search-control>
```
Comment (100stacks) [inline src/elements/hx-search-control/index.spec.js:11]: Please see if this works by moving the `<hx-search-control>` above the code block:

```html
const mockup = `
        <hx-search-control>
          <input
            id="demoSearch"
            type="search"
          />
          <button
            type="button"
            class="hxClear"
            aria-label="Clear search"
            hidden>
            <hx-icon type="times"></hx-icon>
          </button>
          <hx-search></hx-search>
          <label
            for="demoSearch">
            Username
          </label>
        </hx-search-control>`;
```

Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.spec.js (hunk 6 above)
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.spec.js (hunk 5 above)
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.js (hunk 1 above)
Comment (100stacks) [reply src/elements/hx-search-control/index.js:16]: Please see PR #717

Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.js]: Changes have been done, please review.

File: src/elements/hx-search-control/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+            setTimeout(() => fragment.dispatchEvent(customEvent));
+            const evt = await oneEvent(fragment, 'change');
+
+            expect(evt).to.equal(customEvent);
+            expect(evt.detail).to.equal(detail);
+        });
+
+    });
+});
```
Comment (100stacks) [inline src/elements/hx-search-control/index.spec.js:98]: Please add a newline at the EOF.

File: src/elements/hx-search-control/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
+import { fixture, expect, oneEvent } from '@open-wc/testing';
+
+/**
+ * <hx-search-control> component tests
+ *
+ * @type HXSearchControlElement
+ *
+ */
+describe('<hx-search-control> component tests', () => {
```
Comment (100stacks) [inline src/elements/hx-search-control/index.spec.js:9]: Please add a test for the `controlElement` method:

```js
    get controlElement () {
        return this.querySelector('input[type="search"]');
    }
```

As we discussed today, you can get the control element value using:

```js
const ctrl = fragment.controlElement;
```

File: src/elements/hx-search-control/index.spec.js (hunk 8 above)
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js:98]: Changes have been done, please review.

File: src/elements/hx-search-control/index.spec.js (hunk 9 above)
Comment (lalithkarikelli) [reply src/elements/hx-search-control/index.spec.js:9]: Changes have been done, please review.

**Pull request merged**